portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/provider/converter/AnyToAnyExporter.java

```diff
@@ -0,0 +1,54 @@
+package org.datatransferproject.spi.transfer.provider.converter;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+import org.datatransferproject.spi.transfer.provider.ExportResult;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.types.common.ExportInformation;
+import org.datatransferproject.types.common.models.ContainerResource;
+import org.datatransferproject.types.transfer.auth.AuthData;
+
+/**
+ * Allows flexible bridging between adapters of different types with compatible functionality.
+ *
+ * @param <From> The container type supported by the available exporter.
+ * @param <To>   The container type that's desired.
+ */
+public class AnyToAnyExporter<
+    AD extends AuthData,
+    From extends ContainerResource,
+    To extends ContainerResource> implements Exporter<AD, To> {
+
+  private final Exporter<AD, From> exporter;
+  private final Function<From, To> converter;
+  private final Function<ContainerResource, ContainerResource> inputConverter;
+
+  /**
+   * @param exporter       existing exporter
+   * @param converter      function converting between the existing and desired
+   *                       containers.
+   * @param inputConverter converter that's used to adapt the export information that goes into the
+   *                       export call. It's required because exporters often support various
+   *                       container resources as part of export information. E.g. photo exporters
+   *                       support DateRangeContainerResource, while media adapters might only
+   *                       support MediaContainerResource. Oftentimes, Function.identity() is
+   *                       sufficient.
+   */
+  public AnyToAnyExporter(Exporter<AD, From> exporter, Function<From, To> converter,
+      Function<ContainerResource, ContainerResource> inputConverter) {
```

Comment:
Unfortunately, no. That's because an `Exporter<PhotosContainerResource>` takes `ExportInformation<ContainerResource>` as input to its `export` method. This container resource can be anything at all, e.g. `IdOnlyContainerResource`, `DateRangeContainerResource`, `PhotosContainerResource`, or even some private resource etc. For some reason, we're not enforcing strict typing in this case.

The `From` and `To` exporters might support different inputs and still be made compatible via the `AnyToAnyExporter`. For example, converting from `PhotosExporter` to `MediaExporter` might involve passing `Id` and `DateRange` resources unchanged and converting the `PhotosContainerResource` to `MediaContainerResource`.